routes/api.js

```diff
@@ -1,29 +1,117 @@
-    next(); 
-});
+var timer = config.get('General', 'badge_override_interval')
 
 router.post('/user-access', function(req, res) {
-    var stationState = 'Enabled';
-    if(!req.body || req.body.length === 0 || req.body.length > 200) {
-        res.set({
-            'Content-Typer': 'text/plain',
-            'Station-State': stationState = 'Disabled'
-        });
-        res.status(403).send('Forbidden');
-    } else {
-        res.set({
-            'Content-Type': 'text/plain',
-            'User-Id-String': req.body.id,
-            'Station-State': stationState
-        });
-        res.status(200).send('OK');
-    }        
+	if(!req.body || req.body.length === 0 || req.body.length > 200) {
+		// body sanity check
+		res.set({
+				'Content-Type': 'text/plain',
+				'Station-State': 'Disabled'
+		});
+		res.sendStatus(403);
+	};
+
+	if(!req.headers.badge || !req.headers.sid || !req.headers['station-state']) {
+		// check the necessary headers are provided for future processing
+		res.sendStatus(403);
+	};
+
+	// TODO check if user is already logged in to another machine
+	// the piece can be extracted because if the user is on another
+	// machine then there is no need to perform all the checks
+	// this will *might* require a module that contains all users
+	// currently loogged into a machine.
+	
+	var headerObj = req.headers;
+
+	db.getStation(headerObj.sid).then(function(station) {
+		//check if machine is in database
+		!station ? res.sendStatus(403) : console.log(station);
+		db.validateUser(headerObj.badge).then(function(user) {
+			//check if user accesing machine exist
+		 	!user ? res.sendStatus(403) : console.log(user);
+			if (user.dataValues.status === 'Admin') {
+				!user.dataValues.status ? res.sendStatus(503) : console.log(user.dataValues.status)
+				// check cache to see if there is a user on the machine
+				var userWaiting = cache.getMachine(headerObj.sid, timer);
+				if (userWaiting === null && headerObj['station-state'] === 'disabled') {
+					//there is no user in cache and the machine is 'disabled'
+					//put the admin on the machine	
+					// call station module HERE //to update station with current user and change to enabled
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'enabled',
+						'badge':  headerObj.badge
+					});
+					res.sendStatus(200);
+				} else if (headerObj['station-state'] === 'enabled') {
+					//assumption is the admin is logging of from the machine
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'disabled'
+					});
+					res.sendStatus(200);
+				} else {
+					// grant privilages to the user	
+					db.grantPrivileges(userWaiting, headerObj.sid).then(function(result) {
+						cache.delete(headerObj.sid);
+						//call station module HERE //to update station with current user and change to enabled
+						res.set("Connection", "close");
+						!result ? res.sendStatus(403) : res.set({
```

Comment:
I dun't understand what's happening here?
it looks like we're sending a 403 response if `response` is not null which, if `db.grantPrivileges` works as it should, is always. Even if there's an error with modifying the DB, that function should be returning a JSON object that contains an error message (see the `<Object:statusCode>` JSON schema in the DatabaseRequestAnalysis.xlsx doc in g-drive).
This block should only be returning a 200, or possibly a 500 if there's some kind of error with the database and we can't fulfill the request as we should.